File: anosAnteriores/_materialBase/1.1 - argumentosConsole/desafio.cpp
#include <iostream>
#include <string>
#include <fstream>
using namespace std;

bool procurarPalavraNoAquivo(string palavra, char *nomeArquivo) {
    ifstream procuradorArquivo; //tipo de arquivo para leitura
    procuradorArquivo.open(nomeArquivo); 

    if (!procuradorArquivo) {
        cout << nomeArquivo << " eh um arquivo nao localizado." << endl;
        return false;
    }
    	
	//le o arquivo capturando as frases
	string linha;
	while (!procuradorArquivo.eof()) {
		getline(procuradorArquivo,linha); //lendo a linha inteira
		if (linha.find(palavra) != string::npos) {
            procuradorArquivo.close();
            return true;
        }
	}
	procuradorArquivo.close();
    return false;
}

int main(int quantidadeArgumentos, char** vetorArgumentos) {
    
    if (quantidadeArgumentos != 3) {
        cout << "Quantidade de argumentos invalida\n";
        exit(0);
    }

    cout << "Arquivo de busca: " << vetorArgumentos[1] << endl;
    cout << "Palavra de busca: " << vetorArgumentos[2] << endl;

    if (procurarPalavraNoAquivo(vetorArgumentos[2], vetorArgumentos[1])) {
        cout << "Palavra encontrada\n";
    } else {
        cout << "Palavra nao encontrada\n";
    }

    return 1;
}

File: anosAnteriores/_materialBase/1.1 - argumentosConsole/desafio.py
def procurar_palavra_no_arquivo(palavra, nome_arquivo):
    try:
        with open(nome_arquivo, 'r') as arquivo:
            for linha in arquivo:
                if palavra in linha:
                    return True
    except FileNotFoundError:
        print(nome_arquivo, "é um arquivo não encontrado.")
    return False

def main():
    import sys
    
    if len(sys.argv) != 3:
        print("Quantidade de argumentos inválida")
        sys.exit(0)

    nome_arquivo = sys.argv[1]
    palavra = sys.argv[2]

    print("Arquivo de busca:", nome_arquivo)
    print("Palavra de busca:", palavra)

    if procurar_palavra_no_arquivo(palavra, nome_arquivo):
        print("Palavra encontrada")
    else:
        print("Palavra não encontrada")

if __name__ == "__main__":
    main()
